### ISSUE TITLE
Conflict implementation of memfile_istream tellg and seekg functions

### ISSUE THREAD

**Issue by macxfadz** [shaovoon/simplebinstream#4]
Hi,
In **class memfile_istream**  these member functions has been defined:
```
 long tellg(std::FILE* input_file_ptr) const;
void seekg(std::FILE* input_file_ptr, long pos);
void seekg(std::FILE* input_file_ptr, long offset, int way);
```
problem is user has to provide `FILE` handler explicitly as an argument to above functions.
Since `memfile_istream` is constructed by given file name user intend to handle data(binary content of the file) through **_memfile_istream_** , 
```
void memfile_istream::open(const char * file)
		{
			close();
#ifdef _MSC_VER
			std::FILE* input_file_ptr = nullptr;
			fopen_s(&input_file_ptr, file, "rb");
#else
			std::FILE* input_file_ptr = std::fopen(file, "rb");
#endif
			compute_length(input_file_ptr);
			m_arr = new uint8_t[m_size];
			std::fread(m_arr, m_size, 1, input_file_ptr);
			fclose(input_file_ptr);
		}
```
Here you opened file handler and close after filling `m_arr` array. Problem is  you separated the file handler from `memfile_stream` object,Since `tellg` and `seekg` are member functions ,they should relates with file handler-`input_file_ptr`   or `m_arr` ,unless they act like general utility functions.
therefore `
std::FILE* input_file_ptr`  should be member of `class memfile_istream` is n't it?

**Comment by shaovoon** [comment]
Thanks for reporting this problem. I have just fixed it on the master branch.

**Comment by macxfadz** [comment]
@shaovoon  when will you plan to publish a new release? :)

**Comment by shaovoon** [comment]
I just published version 1.0.1 an hour ago.
